src/Utils/schedules.jl

```diff
@@ -67,9 +69,8 @@ function (schedule::TimeInterval)(model)
     if t >= t★
         if schedule.actuations < typemax(Int)
             schedule.actuations += 1
-        else
-            schedule.first_actuation_time = t★
-            schedule.actuations = 1
+        else # re-initialize the schedule to t★
+            initialize!(schedule, t★)
```

Comment:
@glwagner, does this automatically propagate through to `AveragedTimeInterval`?

We're still finding issues with our `AveragedTimeInterval` diagnostics after picking up from a checkpoint. Does something need to be modified here:
https://github.com/CliMA/Oceananigans.jl/blob/abb66e32d333562dd9aaeb7dd2ed8fac5e781368/src/OutputWriters/windowed_time_average.jl#L89-L96

Reply:
More work is needed for `AveragedTimeInterval`. We have to design an interface for checkpointing aspects of the simulation itself, not just the `model`. Basically a time-average is a callback that has a state (like the model) and that state has to be checkpointed and restored in a similar manner to the model state.

Reply:
I suggest opening a new issue to discuss this.

Reply:
that might work! I would test it with a simple example though first!

Reply:
Yep, we're working on that and will try to add a MWE to the issue.